### DISCUSSION TITLE
Search failed (cannot unpack non-iterable NoneType object)

### DISCUSSION THREAD

**Discussion by SavelevGeo** [semiautomaticgit/SemiAutomaticClassificationPlugin#363] [General]
After installing new version cannot run a search of products

The last messages, occurring around search attempt are

```
ERROR|2023-11-01T16:17:23.778|multiprocess_manager|multi_download_file|2317|Error proc 0-cannot unpack non-iterable NoneType object
ERROR|2023-11-01T16:17:23.779|download_products|query_sentinel_2_database|357|error: search failed
DEBUG|2023-11-01T16:17:23.827|ui_utils|remove_progress_bar|257|end progress: None
```

the full log:
[scp.log.txt](https://github.com/semiautomaticgit/SemiAutomaticClassificationPlugin/files/13227411/scp.log.txt)

**Comment by semiautomaticgit** [comment]
Hi @SavelevGeo ,
I think it is a server issue when the extent of the query is small.

If you run your query in the browser from this [link](https://datahub.creodias.eu/odata/v1/Products?&$filter=contains(Name,%27S2%27)%20and%20(startswith(Name,%27S2%27)%20and%20(Attributes/OData.CSC.StringAttribute/any(att:att/Name%20eq%20%27instrumentShortName%27%20and%20att/OData.CSC.StringAttribute/Value%20eq%20%27MSI%27)%20and%20Attributes/OData.CSC.DoubleAttribute/any(att:att/Name%20eq%20%27cloudCover%27%20and%20att/OData.CSC.DoubleAttribute/Value%20le%20100)))%20and%20ContentDate/Start%20ge%202022-11-01T00:00:00.000Z%20and%20ContentDate/Start%20lt%202023-11-01T21:42:55.721Z%20and%20OData.CSC.Intersects(area=geography%27SRID=4326;POLYGON%20((30.215749730569%2059.972316106039,30.215749730569%2059.911163859208,30.366809710932%2059.911163859208,30.366809710932%2059.972316106039,30.215749730569%2059.972316106039))%27)&$orderby=ContentDate/Start%20asc&$expand=Attributes&$count=True&$top=20&$skip=0) it takes a really long time to complete, and the SCP connection times out.

But if you try with a larger area as from this [link](https://datahub.creodias.eu/odata/v1/Products?&$filter=contains(Name,%27S2%27)%20and%20(startswith(Name,%27S2%27)%20and%20(Attributes/OData.CSC.StringAttribute/any(att:att/Name%20eq%20%27instrumentShortName%27%20and%20att/OData.CSC.StringAttribute/Value%20eq%20%27MSI%27)%20and%20Attributes/OData.CSC.DoubleAttribute/any(att:att/Name%20eq%20%27cloudCover%27%20and%20att/OData.CSC.DoubleAttribute/Value%20le%20100)))%20and%20ContentDate/Start%20ge%202022-11-01T00:00:00.000Z%20and%20ContentDate/Start%20lt%202023-11-01T21:42:55.721Z%20and%20OData.CSC.Intersects(area=geography%27SRID=4326;POLYGON%20((29.889110450471%2060.087627981762,29.889110450471%2059.779703368819,30.864788914159%2059.779703368819,30.864788914159%2060.087627981762,29.889110450471%2060.087627981762))%27)&$orderby=ContentDate/Start%20asc&$expand=Attributes&$count=True&$top=20&$skip=0) the result is much faster.

So the solution is to search with a larger area.

**Comment by semiautomaticgit** [comment]
Hello @SavelevGeo , I'm moving this to discussion as it seems a server issue. 
Of course, please open a new ticket in case you get other error messages.
